open_spiel/games/universal_poker_test.cc

```diff
@@ -234,6 +234,167 @@ void ChumpPolicyTests() {
   }
 }
 
+void FullNLBettingTests() {
+  // Test #1
+  // Checks min raising functionality.
+  std::shared_ptr<const Game> game = LoadGame(
+      "universal_poker(betting=nolimit,"
+                      "numPlayers=2,"
+                      "numRounds=4,"
+                      "blind=2 1,"
+                      "firstPlayer=2 1 1 1,"
+                      "numSuits=4,"
+                      "numRanks=13,"
+                      "numHoleCards=2,"
+                      "numBoardCards=0 3 1 1,"
+                      "stack=20 20,"
+                      "bettingAbstraction=fullgame)");
+  std::unique_ptr<State> state = game->NewInitialState();
+  while (state->IsChanceNode())
+    state->ApplyAction(state->LegalActions()[0]);  // deal hole cards
+  // assert all raise increments are valid
+  for (int i = 3; i < 12; ++i)
+    SPIEL_CHECK_TRUE(absl::c_linear_search(state->LegalActions(), i));
+  SPIEL_CHECK_FALSE(absl::c_linear_search(state->LegalActions(), 12));
+  state->ApplyAction(1);  // call big blind
+  state->ApplyAction(1);  // check big blind
+  for (int i = 0; i < 3; ++i)
+    state->ApplyAction(state->LegalActions()[0]);  // deal flop
+  // assert all raise increments are valid
+  for (int i = 3; i < 12; ++i)
+    SPIEL_CHECK_TRUE(absl::c_linear_search(state->LegalActions(), i));
+  // each player keeps min raising until one is all in
+  for (int i = 3; i < 12; ++i)
+    state->ApplyAction(i);
+  state->ApplyAction(1);  // call last raise
+  state->ApplyAction(state->LegalActions()[0]);  // deal turn
+  state->ApplyAction(state->LegalActions()[0]);  // deal river
+  SPIEL_CHECK_EQ(state->Returns()[0], state->Returns()[1]);  // hand is a draw
+  SPIEL_CHECK_TRUE(absl::StrContains(state->ToString(),
+      "ACPC State: STATE:0:cc/r4r6r8r10r12r14r16r18r20c//"
+      ":2c2d|2h2s/3c3d3h/3s/4c"));
+
+  // Test #2
```

Comment:
Shouldn't these separate tests be separate test functions?

Reply:
Makes sense, didn't know which was preferred. It's updated.